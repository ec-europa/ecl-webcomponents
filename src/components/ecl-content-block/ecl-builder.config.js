const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, '../../../node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  scripts: [
    {
      entry: path.resolve(nodeModules, '@ecl/vanilla-component-content-block/content-block.js'),
      dest: path.resolve(outputFolder, 'scripts/ecl-content-block-vanilla.js'),
      options: {
        banner,
        moduleName: 'ECL',
        includePaths,
        sourceMap: false,
      },
    },
  ],
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-content-block-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-content-block-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-content-block-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-content-block-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
  ],
};
