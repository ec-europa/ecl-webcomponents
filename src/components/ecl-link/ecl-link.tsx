import { Component, h, Prop, Element} from '@stencil/core';

@Component({
  tag: 'ecl-link',
  styleUrls: {
    ec: './build/styles/ecl-link-ec.css',
    eu: './build/styles/ecl-link-eu.css',
  },
  shadow: false,
  scoped: false,
  assetsDirs: ['build'],
})

export class EclLink {
  @Element() el: HTMLElement;
  @Prop() theme: string = 'ec';
  @Prop() path: string;
  @Prop() styleClass: string;
  @Prop() variant: string = 'default';
  @Prop() titleAttr: string;
  @Prop() external: boolean = false;

  getClass(): string {
    const styleClasses = [
      `ecl-link`,
      `ecl-link--${this.variant}`,
      this.styleClass
    ];

    if (this.el.querySelector('ecl-icon')) {
      styleClasses.push('ecl-link--icon');
      if (this.el.querySelector('ecl-icon').getAttribute('slot') === 'icon-after') {
        styleClasses.push('ecl-link--icon-after');
      } else {
        styleClasses.push('ecl-link--icon-before');
      }
    }

    return styleClasses.join(' ');
  }

  componentDidRender() {
    if (this.el.querySelector('ecl-icon')) {
      this.el.querySelector('ecl-icon svg').classList.add('ecl-link__icon');
      if (this.el.querySelector('ecl-icon').getAttribute('slot') === 'icon-after') {
        this.el.querySelector('ecl-icon svg').classList.add('ecl-link__icon--after');
      } else {
        this.el.querySelector('ecl-icon svg').classList.add('ecl-link__icon--before');
      }
    }
  }

  getExternal() {
    return (
      <ecl-icon 
        icon="external"
        style-class={`ecl-link__icon ecl-link__icon--after sc-ecl-link-${this.theme}`}
        size = "2xs"
      >
      </ecl-icon>
    )  
  }

  render() {
    return (
      <a
        class={this.getClass()}
        href={this.path}
        title={this.titleAttr}
      >
        <slot name="icon-before"></slot>
        <slot></slot>
        <slot name="icon-after">{ this.external ? this.getExternal() : '' }</slot>
      </a>
    );
  }
}
