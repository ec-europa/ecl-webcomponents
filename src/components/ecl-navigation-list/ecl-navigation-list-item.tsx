import { Component, h, Prop} from '@stencil/core';

@Component({
  tag: 'ecl-navigation-list-item',
  shadow: false,
  scoped: true,
})

export class EclNavigationListItem {
  @Prop() theme: string = 'ec';
  @Prop() itemTitle: string;
  @Prop() image: string;
  @Prop() imageAlt: string;
  @Prop() styleClass: string;
  @Prop() border: boolean = true;


  getClass(): string {
    const styleClasses = [
      `sc-ecl-navigation-list-${this.theme}`,
      `ecl-navigation-list__item`,
      this.styleClass
    ];

    if (!this.border) {
      styleClasses.push('ecl-navigation-list__item--no-border');
    }

    return styleClasses.join(' ');
  }

  getImgClass(): string {
    const imgClasses = [
      `sc-ecl-navigation-list-${this.theme}`,
      `ecl-navigation-list__image`,
    ];

    return imgClasses.join(' ');
  }

  getImgAttr() {
    const attrs = {
      role: 'img',
    }

    if (this.imageAlt) {
      attrs['aria-label'] = this.imageAlt;
    }

    return attrs;
  }

  render() {
    return (
      <div class={this.getClass()}>
      { this.image ?
        <div
          class={this.getImgClass()}
          style={{ backgroundImage: "url(" + this.image + ")"}}
          {...this.getImgAttr()}
        ></div> : ''
      }
        <slot></slot>
      </div>
    );
  }
}