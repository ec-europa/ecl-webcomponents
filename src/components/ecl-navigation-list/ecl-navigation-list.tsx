import { Component, h, Prop, Element} from '@stencil/core';

@Component({
  tag: 'ecl-navigation-list',
  styleUrls: {
    ec: './build/styles/ecl-navigation-list-ec.css',
    eu: './build/styles/ecl-navigation-list-eu.css',
  },
  shadow: false,
  scoped: true,
})

export class EclNavigationList {
  @Element() el: HTMLElement;
  @Prop() theme: string = 'ec';
  @Prop() column: number = 2;
  @Prop() styleClass: string;

  getClass(): string {
    const styleClasses = [
      `ecl-navigation-list`,
      `ecl-navigation-list--col-${this.column}`,
      this.styleClass
    ];

    return styleClasses.join(' ');
  }

  componentDidRender() {
    // Clean the html to make the zebra work,.
    const items = this.el.querySelectorAll('.ecl-navigation-list__item');
    if (items) {
      this.el.querySelector('.ecl-navigation-list').innerHTML = '';
      this.el.querySelector('.ecl-navigation-list').append(...items);
    }
    const blocks = this.el.querySelectorAll('.ecl-content-block');
    if (blocks) {
      blocks.forEach((block) => {
        block.classList.add('ecl-navigation-list__content-block', `sc-ecl-navigation-list-${this.theme}`);
      });
    }
  }

  render() {
    return (
      <div class={this.getClass()}>
        <slot></slot>
      </div>
    );
  }
}
