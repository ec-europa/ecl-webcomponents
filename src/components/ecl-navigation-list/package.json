{
  "name": "@ecl/web-components-navigation-list",
  "author": "European Commission",
  "license": "EUPL-1.2",
  "version": "0.1.0-alpha.0",
  "description": "ECL Message",
  "main": "ecl-navigation-list.tsx",
  "private": true,
  "scripts": {
    "build:styles": "cross-env ecl-builder styles",
    "build": "run-s clean build:*",
    "clean": "rimraf build",
    "prepublish": "npm run build"
  },
  "devDependencies": {
    "@ecl/builder": "3.7.1",
    "@ecl/vanilla-component-navigation-list": "3.7.1"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/planctus/ecl-webcomponents.git"
  },
  "bugs": {
    "url": "https://github.com/planctus/ecl-webcomponents/issues"
  },
  "homepage": "https://github.com/planctus/ecl-webcomponents",
  "keywords": [
    "ecl",
    "europa-component-library",
    "design-system",
    "web-components"
  ]
}
