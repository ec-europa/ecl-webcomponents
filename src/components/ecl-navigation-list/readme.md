# ecl-navigation-list



<!-- Auto Generated Below -->


## Properties

| Property     | Attribute     | Description | Type      | Default     |
| ------------ | ------------- | ----------- | --------- | ----------- |
| `border`     | `border`      |             | `boolean` | `true`      |
| `image`      | `image`       |             | `string`  | `undefined` |
| `imageAlt`   | `image-alt`   |             | `string`  | `undefined` |
| `itemTitle`  | `item-title`  |             | `string`  | `undefined` |
| `styleClass` | `style-class` |             | `string`  | `undefined` |
| `theme`      | `theme`       |             | `string`  | `'ec'`      |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
