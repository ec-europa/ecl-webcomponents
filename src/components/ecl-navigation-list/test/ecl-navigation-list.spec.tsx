import { newSpecPage } from '@stencil/core/testing';
import { EclNavigationList } from '../ecl-navigation-list';
import { EclNavigationListItem } from '../ecl-navigation-list-item';
import { EclContentBlock } from '../../ecl-content-block/ecl-content-block';
import { EclLink } from '../../ecl-link/ecl-link';

describe('ecl-navigation-list', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclNavigationList, EclNavigationListItem, EclContentBlock, EclLink],
      html: `<ecl-navigation-list
        theme="ec"
        column="2"
      >
        <ecl-navigation-list-item
          theme="ec"
          image="https://inno-ecl.s3.amazonaws.com/media/example.htmls/example.html-image.jpg"
          image-alt="Alt text of the image"
          border
        >
          <ecl-content-block
            theme="ec"
            has-title
            has-description
            has-links
            has-secondary-links
          >
            <ecl-link
              theme="ec"
              slot="title"
              path="/example.html"
              variant="standalone"
            >
              Title 1
            </ecl-link>
            <p slot="description">Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
    Phasellus gravida ipsum ut lorem cursus, quis tincidunt sem viverra.
    Nunc vestibulum, mauris quis porta venenatis, justo odio commodo tellus</p>
            <ecl-link
              slot="links"
              theme="ec"
              variant="standalone"
              path="/example.html"
            >
              Primary Link 1
            </ecl-link>
            <ecl-link
              slot="links"
              theme="ec"
              variant="standalone"
              path="/example.html"
            >
              Primary Link 2
            </ecl-link>
            <ecl-link
              slot="links"
              theme="ec"
              variant="standalone"
              path="/example.html"
            >
              Primary Link 3
            </ecl-link>
            <ecl-link
              slot="links"
              theme="ec"
              variant="standalone"
              path="/example.html"
            >
              Primary Link 4
            </ecl-link>
            <ecl-link
              slot="links-secondary"
              theme="ec"
              variant="standalone"
              path="/example.html"
            >
              Secondary Link 1
            </ecl-link>
            <ecl-link
              slot="links-secondary"
              theme="ec"
              variant="standalone"
              path="/example.html"
            >
              Secondary Link 2
            </ecl-link>
          </ecl-content-block>
        </ecl-navigation-list-item>
        <ecl-navigation-list-item
          theme="ec"
          image="https://inno-ecl.s3.amazonaws.com/media/example.htmls/example.html-image.jpg"
          border
          image-alt="Alt text of the image"
        >
          <ecl-content-block
            theme="ec"
            has-title
            has-description
            has-links
            has-secondary-links
          >
            <ecl-link
              theme="ec"
              slot="title"
              path="/example.html"
              variant="standalone"
            >
              Title 2
            </ecl-link>
            <p slot="description">Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
    Phasellus gravida ipsum ut lorem cursus, quis tincidunt sem viverra.
    Nunc vestibulum, mauris quis porta venenatis, justo odio commodo tellus</p>
            <ecl-link
              slot="links"
              theme="ec"
              variant="standalone"
              path="/example.html"
            >
              Primary Link 1
            </ecl-link>
            <ecl-link
              slot="links"
              theme="ec"
              variant="standalone"
              path="/example.html"
            >
              Primary Link 2
            </ecl-link>
            <ecl-link
              slot="links"
              theme="ec"
              variant="standalone"
              path="/example.html"
            >
              Primary Link 3
            </ecl-link>
            <ecl-link
              slot="links"
              theme="ec"
              variant="standalone"
              path="/example.html"
            >
              Primary Link 4
            </ecl-link>
            <ecl-link
              slot="links-secondary"
              theme="ec"
              variant="standalone"
              path="/example.html"
            >
              Secondary Link 1
            </ecl-link>
            <ecl-link
              slot="links-secondary"
              theme="ec"
              variant="standalone"
              path="/example.html"
            >
              Secondary Link 2
            </ecl-link>
          </ecl-content-block>
        </ecl-navigation-list-item>
        <ecl-navigation-list-item
          theme="ec"
          image="https://inno-ecl.s3.amazonaws.com/media/example.htmls/example.html-image.jpg"
          border
          image-alt="Alt text of the image"
        >
          <ecl-content-block
            theme="ec"
            has-title
            has-description
            has-links
            has-secondary-links
          >
            <ecl-link
              theme="ec"
              slot="title"
              path="/example.html"
              variant="standalone"
            >
              Title 3
            </ecl-link>
            <p slot="description">Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
    Phasellus gravida ipsum ut lorem cursus, quis tincidunt sem viverra.
    Nunc vestibulum, mauris quis porta venenatis, justo odio commodo tellus</p>
            <ecl-link
              slot="links"
              theme="ec"
              variant="standalone"
              path="/example.html"
            >
              Primary Link 1
            </ecl-link>
            <ecl-link
              slot="links"
              theme="ec"
              variant="standalone"
              path="/example.html"
            >
              Primary Link 2
            </ecl-link>
            <ecl-link
              slot="links"
              theme="ec"
              variant="standalone"
              path="/example.html"
            >
              Primary Link 3
            </ecl-link>
            <ecl-link
              slot="links"
              theme="ec"
              variant="standalone"
              path="/example.html"
            >
              Primary Link 4
            </ecl-link>
            <ecl-link
              slot="links-secondary"
              theme="ec"
              variant="standalone"
              path="/example.html"
            >
              Secondary Link 1
            </ecl-link>
            <ecl-link
              slot="links-secondary"
              theme="ec"
              variant="standalone"
              path="/example.html"
            >
              Secondary Link 2
            </ecl-link>
          </ecl-content-block>
        </ecl-navigation-list-item>
        <ecl-navigation-list-item
          theme="ec"
          image="https://inno-ecl.s3.amazonaws.com/media/example.htmls/example.html-image.jpg"
          border
          image-alt="Alt text of the image"
        >
          <ecl-content-block
            theme="ec"
            has-title
            has-description
            has-links
            has-secondary-links
          >
            <ecl-link
              theme="ec"
              slot="title"
              path="/example.html"
              variant="standalone"
            >
              Title 4
            </ecl-link>
            <p slot="description">Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
    Phasellus gravida ipsum ut lorem cursus, quis tincidunt sem viverra.
    Nunc vestibulum, mauris quis porta venenatis, justo odio commodo tellus</p>
            <ecl-link
              slot="links"
              theme="ec"
              variant="standalone"
              path="/example.html"
            >
              Primary Link 1
            </ecl-link>
            <ecl-link
              slot="links"
              theme="ec"
              variant="standalone"
              path="/example.html"
            >
              Primary Link 2
            </ecl-link>
            <ecl-link
              slot="links"
              theme="ec"
              variant="standalone"
              path="/example.html"
            >
              Primary Link 3
            </ecl-link>
            <ecl-link
              slot="links"
              theme="ec"
              variant="standalone"
              path="/example.html"
            >
              Primary Link 4
            </ecl-link>
            <ecl-link
              slot="links-secondary"
              theme="ec"
              variant="standalone"
              path="/example.html"
            >
              Secondary Link 1
            </ecl-link>
            <ecl-link
              slot="links-secondary"
              theme="ec"
              variant="standalone"
              path="/example.html"
            >
              Secondary Link 2
            </ecl-link>
          </ecl-content-block>
        </ecl-navigation-list-item>
        <ecl-navigation-list-item
          theme="ec"
          image="https://inno-ecl.s3.amazonaws.com/media/example.htmls/example.html-image.jpg"
          border
          image-alt="Alt text of the image"
        >
          <ecl-content-block
            theme="ec"
            has-title
            has-description
            has-links
            has-secondary-links
          >
            <ecl-link
              theme="ec"
              slot="title"
              path="/example.html"
              variant="standalone"
            >
              Title 5
            </ecl-link>
            <p slot="description">Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
    Phasellus gravida ipsum ut lorem cursus, quis tincidunt sem viverra.
    Nunc vestibulum, mauris quis porta venenatis, justo odio commodo tellus</p>
            <ecl-link
              slot="links"
              theme="ec"
              variant="standalone"
              path="/example.html"
            >
              Primary Link 1
            </ecl-link>
            <ecl-link
              slot="links"
              theme="ec"
              variant="standalone"
              path="/example.html"
            >
              Primary Link 2
            </ecl-link>
            <ecl-link
              slot="links"
              theme="ec"
              variant="standalone"
              path="/example.html"
            >
              Primary Link 3
            </ecl-link>
            <ecl-link
              slot="links"
              theme="ec"
              variant="standalone"
              path="/example.html"
            >
              Primary Link 4
            </ecl-link>
            <ecl-link
              slot="links-secondary"
              theme="ec"
              variant="standalone"
              path="/example.html"
            >
              Secondary Link 1
            </ecl-link>
            <ecl-link
              slot="links-secondary"
              theme="ec"
              variant="standalone"
              path="/example.html"
            >
              Secondary Link 2
            </ecl-link>
          </ecl-content-block>
        </ecl-navigation-list-item>
      </ecl-navigation-list>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});
