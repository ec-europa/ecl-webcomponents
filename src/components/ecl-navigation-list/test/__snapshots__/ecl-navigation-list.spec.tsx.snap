// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-navigation-list renders 1`] = `
<ecl-navigation-list column="2" theme="ec">
  <!---->
  <div class="ecl-navigation-list ecl-navigation-list--col-2">
    <div class="ecl-navigation-list__item sc-ecl-navigation-list-ec">
      <ecl-picture image="https://inno-ecl.s3.amazonaws.com/media/examples/example.html-image.jpg" imagealt="Alt text of the image" imgclass="sc-ecl-navigation-list-ec ecl-navigation-list__image" styleclass="ecl-navigation-list__picture sc-ecl-navigation-list-ec"></ecl-picture>
      <ecl-content-block has-description="" has-links="" has-secondary-links="" has-title="" theme="ec">
        <!---->
        <div class="ecl-content-block ecl-navigation-list__content-block sc-ecl-navigation-list-ec">
          <h1 class="ecl-content-block__title">
            <ecl-link path="/example.html" slot="title" theme="ec" variant="standalone">
              <!---->
              <a class="ecl-link ecl-link--standalone" href="/example.html">
                Title 1
                <slot-fb name="icon-after"></slot-fb>
              </a>
            </ecl-link>
          </h1>
          <div class="ecl-content-block__description">
            <p slot="description">
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus gravida ipsum ut lorem cursus, quis tincidunt sem viverra. Nunc vestibulum, mauris quis porta venenatis, justo odio commodo tellus
            </p>
          </div>
          <div class="ecl-content-block__link-container">
            <ul class="ecl-content-block__link-list">
              <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
                <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
                  Primary Link 1
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </li>
              <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
                <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
                  Primary Link 2
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </li>
              <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
                <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
                  Primary Link 3
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </li>
              <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
                <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
                  Primary Link 4
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </li>
            </ul>
            <ul class="ecl-content-block__link-list">
              <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
                <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
                  Secondary Link 1
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </li>
              <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
                <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
                  Secondary Link 2
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </li>
            </ul>
          </div>
        </div>
      </ecl-content-block>
    </div>
    <div class="ecl-navigation-list__item sc-ecl-navigation-list-ec">
      <ecl-picture image="https://inno-ecl.s3.amazonaws.com/media/examples/example.html-image.jpg" imagealt="Alt text of the image" imgclass="sc-ecl-navigation-list-ec ecl-navigation-list__image" styleclass="ecl-navigation-list__picture sc-ecl-navigation-list-ec"></ecl-picture>
      <ecl-content-block has-description="" has-links="" has-secondary-links="" has-title="" theme="ec">
        <!---->
        <div class="ecl-content-block ecl-navigation-list__content-block sc-ecl-navigation-list-ec">
          <h1 class="ecl-content-block__title">
            <ecl-link path="/example.html" slot="title" theme="ec" variant="standalone">
              <!---->
              <a class="ecl-link ecl-link--standalone" href="/example.html">
                Title 2
                <slot-fb name="icon-after"></slot-fb>
              </a>
            </ecl-link>
          </h1>
          <div class="ecl-content-block__description">
            <p slot="description">
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus gravida ipsum ut lorem cursus, quis tincidunt sem viverra. Nunc vestibulum, mauris quis porta venenatis, justo odio commodo tellus
            </p>
          </div>
          <div class="ecl-content-block__link-container">
            <ul class="ecl-content-block__link-list">
              <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
                <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
                  Primary Link 1
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </li>
              <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
                <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
                  Primary Link 2
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </li>
              <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
                <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
                  Primary Link 3
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </li>
              <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
                <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
                  Primary Link 4
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </li>
            </ul>
            <ul class="ecl-content-block__link-list">
              <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
                <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
                  Secondary Link 1
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </li>
              <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
                <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
                  Secondary Link 2
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </li>
            </ul>
          </div>
        </div>
      </ecl-content-block>
    </div>
    <div class="ecl-navigation-list__item sc-ecl-navigation-list-ec">
      <ecl-picture image="https://inno-ecl.s3.amazonaws.com/media/examples/example.html-image.jpg" imagealt="Alt text of the image" imgclass="sc-ecl-navigation-list-ec ecl-navigation-list__image" styleclass="ecl-navigation-list__picture sc-ecl-navigation-list-ec"></ecl-picture>
      <ecl-content-block has-description="" has-links="" has-secondary-links="" has-title="" theme="ec">
        <!---->
        <div class="ecl-content-block ecl-navigation-list__content-block sc-ecl-navigation-list-ec">
          <h1 class="ecl-content-block__title">
            <ecl-link path="/example.html" slot="title" theme="ec" variant="standalone">
              <!---->
              <a class="ecl-link ecl-link--standalone" href="/example.html">
                Title 3
                <slot-fb name="icon-after"></slot-fb>
              </a>
            </ecl-link>
          </h1>
          <div class="ecl-content-block__description">
            <p slot="description">
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus gravida ipsum ut lorem cursus, quis tincidunt sem viverra. Nunc vestibulum, mauris quis porta venenatis, justo odio commodo tellus
            </p>
          </div>
          <div class="ecl-content-block__link-container">
            <ul class="ecl-content-block__link-list">
              <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
                <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
                  Primary Link 1
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </li>
              <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
                <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
                  Primary Link 2
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </li>
              <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
                <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
                  Primary Link 3
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </li>
              <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
                <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
                  Primary Link 4
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </li>
            </ul>
            <ul class="ecl-content-block__link-list">
              <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
                <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
                  Secondary Link 1
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </li>
              <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
                <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
                  Secondary Link 2
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </li>
            </ul>
          </div>
        </div>
      </ecl-content-block>
    </div>
    <div class="ecl-navigation-list__item sc-ecl-navigation-list-ec">
      <ecl-picture image="https://inno-ecl.s3.amazonaws.com/media/examples/example.html-image.jpg" imagealt="Alt text of the image" imgclass="sc-ecl-navigation-list-ec ecl-navigation-list__image" styleclass="ecl-navigation-list__picture sc-ecl-navigation-list-ec"></ecl-picture>
      <ecl-content-block has-description="" has-links="" has-secondary-links="" has-title="" theme="ec">
        <!---->
        <div class="ecl-content-block ecl-navigation-list__content-block sc-ecl-navigation-list-ec">
          <h1 class="ecl-content-block__title">
            <ecl-link path="/example.html" slot="title" theme="ec" variant="standalone">
              <!---->
              <a class="ecl-link ecl-link--standalone" href="/example.html">
                Title 4
                <slot-fb name="icon-after"></slot-fb>
              </a>
            </ecl-link>
          </h1>
          <div class="ecl-content-block__description">
            <p slot="description">
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus gravida ipsum ut lorem cursus, quis tincidunt sem viverra. Nunc vestibulum, mauris quis porta venenatis, justo odio commodo tellus
            </p>
          </div>
          <div class="ecl-content-block__link-container">
            <ul class="ecl-content-block__link-list">
              <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
                <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
                  Primary Link 1
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </li>
              <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
                <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
                  Primary Link 2
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </li>
              <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
                <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
                  Primary Link 3
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </li>
              <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
                <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
                  Primary Link 4
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </li>
            </ul>
            <ul class="ecl-content-block__link-list">
              <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
                <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
                  Secondary Link 1
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </li>
              <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
                <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
                  Secondary Link 2
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </li>
            </ul>
          </div>
        </div>
      </ecl-content-block>
    </div>
    <div class="ecl-navigation-list__item sc-ecl-navigation-list-ec">
      <ecl-picture image="https://inno-ecl.s3.amazonaws.com/media/examples/example.html-image.jpg" imagealt="Alt text of the image" imgclass="sc-ecl-navigation-list-ec ecl-navigation-list__image" styleclass="ecl-navigation-list__picture sc-ecl-navigation-list-ec"></ecl-picture>
      <ecl-content-block has-description="" has-links="" has-secondary-links="" has-title="" theme="ec">
        <!---->
        <div class="ecl-content-block ecl-navigation-list__content-block sc-ecl-navigation-list-ec">
          <h1 class="ecl-content-block__title">
            <ecl-link path="/example.html" slot="title" theme="ec" variant="standalone">
              <!---->
              <a class="ecl-link ecl-link--standalone" href="/example.html">
                Title 5
                <slot-fb name="icon-after"></slot-fb>
              </a>
            </ecl-link>
          </h1>
          <div class="ecl-content-block__description">
            <p slot="description">
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus gravida ipsum ut lorem cursus, quis tincidunt sem viverra. Nunc vestibulum, mauris quis porta venenatis, justo odio commodo tellus
            </p>
          </div>
          <div class="ecl-content-block__link-container">
            <ul class="ecl-content-block__link-list">
              <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
                <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
                  Primary Link 1
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </li>
              <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
                <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
                  Primary Link 2
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </li>
              <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
                <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
                  Primary Link 3
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </li>
              <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
                <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
                  Primary Link 4
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </li>
            </ul>
            <ul class="ecl-content-block__link-list">
              <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
                <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
                  Secondary Link 1
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </li>
              <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
                <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
                  Secondary Link 2
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </li>
            </ul>
          </div>
        </div>
      </ecl-content-block>
    </div>
  </div>
</ecl-navigation-list>
`;
